dotnet/src/Extensions/Extensions.UnitTests/TemplateEngine/Prompt/PromptTemplateEngineTests.cs

```diff
@@ -21,6 +21,8 @@
 
 namespace SemanticKernel.Extensions.UnitTests.TemplateEngine.Prompt;
 
+[Obsolete("Use BasicPromptTemplateFactory instead. This will be removed in a future release.")]
+[EditorBrowsable(EditorBrowsableState.Never)]
 public sealed class PromptTemplateEngineTests
```

Comment:
Tests in the Basic template engine should still be applicable to the new code, can we alter them to be `Factory` friendly?

Make sure to test also the composition of different factories.